src/analysis/retail/demonhunter/vengeance/modules/talents/CycleOfBinding.tsx

```diff
@@ -67,103 +40,37 @@ export default class CycleOfBinding extends Analyzer.withDependencies(deps) {
       sigilOfChainsSpell,
     ];
 
-    this.addEventListener(
-      Events.cast.by(SELECTED_PLAYER).spell(sigilOfFlameSpell),
-      this.onSigilOfFlameCast,
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(sigilOfFlameSpell), (event) =>
+      this.onEvent(event, cycleOfBindingNormalizer.getTargetsAffectedBySigilOfFlame),
     );
-    this.addEventListener(
-      Events.cast.by(SELECTED_PLAYER).spell(sigilOfMiserySpell),
-      this.onSigilOfMiseryCast,
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(sigilOfMiserySpell), (event) =>
+      this.onEvent(event, cycleOfBindingNormalizer.getTargetsAffectedBySigilOfMisery),
     );
-    this.addEventListener(
-      Events.cast.by(SELECTED_PLAYER).spell(elysianDecreeSpell),
-      this.onElysianDecreeCast,
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(elysianDecreeSpell), (event) =>
+      this.onEvent(event, cycleOfBindingNormalizer.getTargetsAffectedByElysianDecree),
     );
-    this.addEventListener(
-      Events.cast.by(SELECTED_PLAYER).spell(sigilOfSilenceSpell),
-      this.onSigilOfSilenceCast,
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(sigilOfSilenceSpell), (event) =>
+      this.onEvent(event, cycleOfBindingNormalizer.getTargetsAffectedBySigilOfSilence),
     );
-    this.addEventListener(
-      Events.cast.by(SELECTED_PLAYER).spell(sigilOfChainsSpell),
-      this.onSigilOfChainsCast,
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(sigilOfChainsSpell), (event) =>
+      this.onEvent(event, cycleOfBindingNormalizer.getTargetsAffectedBySigilOfChains),
```

Comment:
this is a pretty good refactor. could be cleaner if the list of getters were indexed by spell so you could just loop over them, but that doesn't exist as-is so not too fussed.